### PR TITLE
fix: swap stringify with custom serializer to speed up renders

### PR DESCRIPTION
**Pull request by grod220**
Follow up to: https://github.com/sushiswap/sushiswap-interface/pull/339

### CODE REVIEW

File: src/state/wallet/hooks.ts [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { useMultipleContractSingleData, useSingleContractMultipleData } from '..
 
 import ERC20_ABI from '../../constants/abis/erc20.json'
 import { Interface } from '@ethersproject/abi'
-import { SUSHI } from './../../config/tokens'
```
Comment (grod220) [inline src/state/wallet/hooks.ts:6]: Unused import

File: src/state/wallet/hooks.ts [hunk 2]
```diff
@@ -77,9 +77,7 @@ export function useTokenBalancesWithLoadingIndicator(
     useMemo(
       () =>
         address && validatedTokens.length > 0
-          ? validatedTokens.reduce<{
-              [tokenAddress: string]: CurrencyAmount<Token> | undefined
```
Comment (grod220) [inline src/state/wallet/hooks.ts:81]: Found that the type of `[tokenAddress: string]: undefined` can't ever be the case given we only add it to object if it has a valid amount. The only two scenarios seem to be an empty object or entries with values.

File: src/state/wallet/hooks.test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { CurrencyAmount, Token } from '@sushiswap/sdk'
```
Comment (grod220) [inline src/state/wallet/hooks.test.ts:1]: Did not see a test runner setup in the repo even though there are tests. Wrote these and installed jest locally to verify they pass. I suppose it's another task to add jest properly to the repo.

File: src/state/wallet/hooks.ts [hunk 4]
```diff
@@ -140,14 +146,11 @@ export function useCurrencyBalance(account?: string, currency?: Currency): Curre
...
 // mimics useAllBalances
-export function useAllTokenBalances(): {
-  [tokenAddress: string]: CurrencyAmount<Token> | undefined
-} {
+export function useAllTokenBalances(): TokenBalancesMap {
   const { account } = useActiveWeb3React()
   const allTokens = useAllTokens()
   const allTokensArray = useMemo(() => Object.values(allTokens ?? {}), [allTokens])
-  const balances = useTokenBalances(account ?? undefined, allTokensArray)
-  return balances ?? {}
```
Comment (grod220) [inline src/state/wallet/hooks.ts:150]: Related to: https://github.com/sushiswap/sushiswap-interface/pull/344/files#r710969747. `useTokenBalances` will never return a falsey value.

**Comment by chillichelli** [comment]
Looks good to me!

**Review by matthewlilley — APPROVED** [review]

**Pull request merged**